src/smspark/bootstrapper.py

```diff
@@ -63,28 +67,42 @@ def bootstrap_history_server(self) -> None:
 
     def copy_aws_jars(self) -> None:
         self.logger.info("copying aws jars")
-        jar_dest = Bootstrapper.SPARK_PATH + "/jars"
         for f in glob.glob("/usr/share/aws/aws-java-sdk/*.jar"):
-            shutil.copyfile(f, os.path.join(jar_dest, os.path.basename(f)))
-        hadoop_aws_jar = "hadoop-aws-2.8.5-amzn-6.jar"
-        jets3t_jar = "jets3t-0.9.0.jar"
-        shutil.copyfile(
-            os.path.join(Bootstrapper.HADOOP_PATH, hadoop_aws_jar), os.path.join(jar_dest, hadoop_aws_jar),
-        )
-        # this jar required for using s3a client
+            shutil.copyfile(f, os.path.join(self.JAR_DEST, os.path.basename(f)))
+        hadoop_aws_jar = self._get_hadoop_jar()
         shutil.copyfile(
-            os.path.join(Bootstrapper.HADOOP_PATH + "/lib", jets3t_jar), os.path.join(jar_dest, jets3t_jar),
+            os.path.join(Bootstrapper.HADOOP_PATH, hadoop_aws_jar), os.path.join(self.JAR_DEST, hadoop_aws_jar)
         )
+
+        self._copy_optional_jars()
         # copy hmclient (glue data catalog hive metastore client) jars to classpath:
         # https://github.com/awslabs/aws-glue-data-catalog-client-for-apache-hive-metastore
         for f in glob.glob("/usr/share/aws/hmclient/lib/*.jar"):
-            shutil.copyfile(f, os.path.join(jar_dest, os.path.basename(f)))
+            shutil.copyfile(f, os.path.join(self.JAR_DEST, os.path.basename(f)))
+
+    def _get_hadoop_jar(self) -> str:
+        for file_name in os.listdir(Bootstrapper.HADOOP_PATH):
+            if file_name.startswith("hadoop-aws") and file_name.endswith(".jar"):
+                self.logger.info(f"Found hadoop jar {file_name}")
+                return file_name
+
+        raise AlgorithmError("Error finding hadoop jar", caused_by=FileNotFoundError())
+
+    def _copy_optional_jars(self) -> None:
+        for jar, jar_path in self.OPTIONAL_JARS.items():
+            if os.path.isfile(os.path.join(jar_path, jar)):
+                self.logger.info(f"Copying optional jar {jar} from {jar_path} to {self.JAR_DEST}")
+                shutil.copyfile(
+                    os.path.join(jar_path, jar), os.path.join(self.JAR_DEST, jar),
+                )
+            else:
+                self.logger.info(f"Optional jar {jar} in {jar_path} does not exist")
```

Comment:
(Optional for now): IMO, we should have a map that only attempts to copy a JAR if we already expect to do so for a particular version:

For example, we'd only expect `my-optional-library.jar` to be there for version `2.4.0`:

```
{"2.4.0": {"/lib/my-optional-library.jar": "/my/jar/destination/path"}}
```

(as in  the previous comment, we should end up doing this at build time instead of runtime)

Reply:
makes sense, will add a todo.

Just that we need a way to get spark version here.